pkg/storage/db.go

```diff
@@ -473,8 +475,125 @@ func (db *DB) Store(ctx context.Context, cloudAssetChanges domain.CloudAssetChan
 		}
 		return err
 	}
-	return tx.Commit()
+	if err = tx.Commit(); err != nil {
+		return err
+	}
+	ver, err := db.GetSchemaVersion(ctx)
+	if err != nil || ver < DualWriteSchemaVersion {
+		return nil
+	}
+	return db.StoreV2(ctx, cloudAssetChanges)
+}
 
+// StoreV2 Currently public to allow testing separately.Storage interface implementation that records to a database using new schema, to replace Store in the future.
+func (db *DB) StoreV2(ctx context.Context, cloudAssetChanges domain.CloudAssetChanges) error {
+	tx, err := db.sqldb.Begin()
+	if err != nil {
+		return err
+	}
+	if err = db.ensureResourceExists(ctx, cloudAssetChanges, tx); err == nil {
+		arnID := resIDFromARN(cloudAssetChanges.ARN)
+		for _, val := range cloudAssetChanges.Changes {
+			for _, ip := range val.PrivateIPAddresses {
+				if strings.EqualFold(added, val.ChangeType) {
```

Comment:
Is it possible for an event to be an "UPDATE" IP address, and would we treat that the same as a DELETE? Or does the data come across as a DELETE plus an ADD? I don't know how AWS Config structures their events re: changes to ip address...

Reply:
The only two types we support are UPDATE and DELETE (basically the if is copy/paste from existing Store). E.g. if an event with anything else for changetype arrived, we'd reject it at the swagger descriptor level as https://github.com/asecurityteam/asset-inventory-api/blob/master/api.yaml#L550 specifically lists ADDED and DELETED as supported event types.